### PR TITLE
docs: improve merge confidence api base url variable doc

### PR DESCRIPTION
**Pull request by mikaello**
## Changes

Improved the documentation for the env variable `RENOVATE_X_MERGE_CONFIDENCE_API_BASE_URL`. I have found it a bit confusing how this works, so I try to document this as I figure it out.

## Context

See https://github.com/mend/renovate-ce-ee/issues/488#issuecomment-2066195342 for some context.

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: docs/usage/self-hosted-experimental.md [hunk 1]
```diff
@@ -134,6 +134,7 @@ Skip initializing `RE2` for regular expressions and instead use Node-native `Reg
 ## `RENOVATE_X_MERGE_CONFIDENCE_API_BASE_URL`
 
 If set, Renovate will query this API for Merge Confidence data.
+If using Mend Renovate Enterprise and a static merge confidence token that you set via `MEND_RNV_MC_TOKEN` (or set `MEND_RNV_MC_TOKEN` to `auto`), you will need to set this variable at the server as well as the workers, if you have specified the token as a [match confidence](https://docs.renovatebot.com/configuration-options/#matchconfidence) `hostRule`, you will only need to set this variable at the workers.
```
Comment (HonkingGoose) [inline docs/usage/self-hosted-experimental.md:137]:
```suggestion

If you use Mend Renovate Enterprise and:

- have a static merge confidence token that you set via `MEND_RNV_MC_TOKEN`
- _or_ set `MEND_RNV_MC_TOKEN` to `auto`

Then you must set this variable at the _server_ and the _workers_.

But if you have specified the token as a [`matchConfidence`](https://docs.renovatebot.com/configuration-options/#matchconfidence) `hostRule`, you only need to set this variable at the _workers_.
```

Changes:

- use a list to split complex clauses
- _emphasize_ the key differences
- apply code styling for the `matchConfidence` config option name
- use the word must to indicate requirements

I'll let a maintainer check the content for technical accuracy.

The link to the doc should be changed to a relative link, I think.

**Review by HonkingGoose — CHANGES_REQUESTED** [review]

**Comment by HonkingGoose** [comment]
Please avoid force-pushing to the PR's branch. This resets the review progress, which makes it harder to see what changed.

We will squash merge this PR, and thus end with one clean commit on the `main` branch. So you do not need to keep your branch tidy. 🙂

**Comment by mikaello** [comment]
> Please avoid force-pushing to the PR's branch. This resets the review progress, which makes it harder to see what changed.

 I see, I will remember this for future contributions. 

I think there is a repo setting to disable the rebase button in PRs. Since it was not disabled I thought this was ok to select that option in the drop down menu for updating the branch.

**Comment by HonkingGoose** [comment]
Thank you for remembering it for future contributions! :smile: 

Hmm. I think that repository setting only limits the type of merges to the `main` branch. In short, it limits what merges a _maintainer_ can select when merging a PR.

That repository setting does _not_ seem to change which options the _user_ sees to sync their _fork_ of `renovate`.

File: docs/usage/self-hosted-experimental.md [hunk 2]
```diff
@@ -134,6 +134,15 @@ Skip initializing `RE2` for regular expressions and instead use Node-native `Reg
 ## `RENOVATE_X_MERGE_CONFIDENCE_API_BASE_URL`
 
 If set, Renovate will query this API for Merge Confidence data.
+
+If you use Mend Renovate Enterprise and:
```
Comment (HonkingGoose) [inline docs/usage/self-hosted-experimental.md:138]:
```suggestion
If you use the Mend Renovate Enterprise Edition (Renovate EE) and:
```

I think `Mend Renovate Enterprise Edition` is the preferred branding of the paid offering? It's what they use on [the Mend.io site, Mend Renovate Enterprise Edition page](https://www.mend.io/renovate-enterprise/).

Also, here's a quote from the [Renovate docs, running Renovate](https://docs.renovatebot.com/getting-started/running/#mend-renovate-community-edition-enterprise-edition):

> Mend Renovate Community Edition (Renovate CE) and Enterprise Edition (Renovate EE) are closed-source offerings of Renovate for self-hosted users.

File: docs/usage/self-hosted-experimental.md [hunk 3]
```diff
@@ -134,6 +134,15 @@ Skip initializing `RE2` for regular expressions and instead use Node-native `Reg
...
+
+- have a static merge confidence token that you set via `MEND_RNV_MC_TOKEN`
+- _or_ set `MEND_RNV_MC_TOKEN` to `auto`
+
+Then you must set this variable at the _server_ and the _workers_.
+
+But if you have specified the token as a [`matchConfidence`](https://docs.renovatebot.com/configuration-options/#matchconfidence) `hostRule`, you only need to set this variable at the _workers_.
 This feature is in private beta.
```
Comment (HonkingGoose) [inline docs/usage/self-hosted-experimental.md:146]:
```suggestion

This feature is in private beta.
```

This sentence needs to be its own paragraph. This sentence has no connection to the previous part about the token.

**Review by HonkingGoose — CHANGES_REQUESTED** [review]

File: docs/usage/self-hosted-experimental.md (hunk 2 above)
Comment (mikaello) [reply docs/usage/self-hosted-experimental.md:138]: Nice clarification!

**Review by rarkins — APPROVED** [review]

**Comment by HonkingGoose** [comment]
@mikaello thank you for improving the docs with your PR! ❤️🥳

**Pull request merged**